2.0 RC4 Arming issue after CMS save reboot

After save & reboot from CMS, motors do not spin after arming.  However stat will still appear during disarm.

In 1.9.x, CMS save & reboot produce ESC last startup tone, but in 2.0 RC4, they are gone.

Reboot in CLI and configurator (MSP I guess) are fine, motors will spin when armed.
I found little difference in code that CLI and MSP delay for 1000ms but CMS only for 200ms

I try to check out RC4 tag as a branch, change it to 1000 and build using docker, but fail with

>$./build.sh MATEKF405
Building target MATEKF405
settings.yaml -> settings_generated.h, settings_generated.c 
%% startup_stm32f40xx.s 
make: *** No rule to make target 'lib/main/CMSIS/CM4/DeviceSupport/ST/STM32F4xx/stm32f4xx_gpio.c', needed by 'obj/main/MATEKF405/stm32f4xx_gpio.o'.  Stop.

I can build the master branch, but seems it is not 2.0RC4

--- a/src/main/cms/cms.c
+++ b/src/main/cms/cms.c
@@ -56,6 +56,7 @@
 #include "config/parameter_group_ids.h"
 
 // For 'ARM' related
+#include "fc/fc_core.h"
 #include "fc/config.h"
 #include "fc/rc_controls.h"
 #include "fc/runtime_config.h"
@@ -758,11 +759,7 @@ long cmsMenuExit(displayPort_t *pDisplay, const void *ptr)
 
         displayResync(pDisplay); // Was max7456RefreshAll(); why at this timing?
 
-        stopMotors();
-        stopPwmAllMotors();
-        delay(200);
-
-        systemReset();
+        fcReboot(false);
     }
 
     DISABLE_ARMING_FLAG(ARMING_DISABLED_CMS_MENU);

--- a/src/main/fc/cli.c
+++ b/src/main/fc/cli.c
@@ -65,6 +65,7 @@ extern uint8_t __config_end;
 #include "drivers/time.h"
 #include "drivers/timer.h"
 
+#include "fc/fc_core.h"
 #include "fc/cli.h"
 #include "fc/config.h"
 #include "fc/controlrate_profile.h"
@@ -2023,15 +2024,7 @@ static void cliRebootEx(bool bootLoader)
     bufWriterFlush(cliWriter);
     waitForSerialPortToFinishTransmitting(cliPort);
 
-    stopMotors();
-    stopPwmAllMotors();
-
-    delay(1000);
-    if (bootLoader) {
-        systemResetToBootloader();
-        return;
-    }
-    systemReset();
+    fcReboot(bootLoader);
 }
 
 static void cliReboot(void)

--- a/src/main/fc/fc_core.c
+++ b/src/main/fc/fc_core.c
@@ -35,6 +35,7 @@
 #include "drivers/serial.h"
 #include "drivers/time.h"
 
+#include "drivers/system.h"
 #include "sensors/sensors.h"
 #include "sensors/diagnostics.h"
 #include "sensors/boardalignment.h"

--- a/src/main/fc/fc_core.c
+++ b/src/main/fc/fc_core.c
@@ -833,6 +833,26 @@ void taskUpdateRxMain(timeUs_t currentTimeUs)
 }
 
 // returns seconds
-float getFlightTime() {
+float getFlightTime()
+{
     return (float)(flightTime / 1000) / 1000;
+}
+
+void fcReboot(bool bootLoader)
+{
+    // stop motor/servo outputs
+    stopMotors();
+    stopPwmAllMotors();
+
+    // extra delay before reboot to give ESCs chance to reset
+    delay(1000);
+
+    if (bootLoader) {
+        systemResetToBootloader();
+    }
+    else {
+        systemReset();
+    }
+
+    while (true);
 }

--- a/src/main/fc/fc_core.h
+++ b/src/main/fc/fc_core.h
@@ -40,3 +40,5 @@ disarmReason_t getDisarmReason(void);
 
 bool isCalibrating(void);
 float getFlightTime();
+
+void fcReboot(bool bootLoader);

--- a/src/main/fc/fc_msp.c
+++ b/src/main/fc/fc_msp.c
@@ -49,6 +49,7 @@
 #include "drivers/timer.h"
 #include "drivers/vtx_common.h"
 
+#include "fc/fc_core.h"
 #include "fc/config.h"
 #include "fc/controlrate_profile.h"
 #include "fc/fc_msp.h"
@@ -187,16 +188,7 @@ static void mspFc4waySerialCommand(sbuf_t *dst, sbuf_t *src, mspPostProcessFnPtr
 static void mspRebootFn(serialPort_t *serialPort)
 {
     UNUSED(serialPort);
-
-    stopMotors();
-    stopPwmAllMotors();
-
-    // extra delay before reboot to give ESCs chance to reset
-    delay(1000);
-    systemReset();
-
-    // control should never return here.
-    while (true) ;
+    fcReboot(false);
 }
 
 static void serializeSDCardSummaryReply(sbuf_t *dst)

